src/Filter/NarrativeFilter.php

```diff
@@ -52,7 +52,9 @@ public function isFeatureMatch(FeatureNode $feature)
      *
      * @param ScenarioInterface $scenario Scenario or Outline node instance
      *
-     * @return bool
+     * @return false This filter is designed to work only with features
+     *
+     * @codeCoverageIgnore
```

Comment:
I guess it doesn't make much sense to have a test that checks that these methods return false. PHPStan in fact would even complain that the test assertion will always be false.

Reply:
@stof for example in this case, if I add a test for it phpstan complains that `assertFalse(false)` is always true (`false` => because of the phpdoc).

If we want to make tests for this, we can change the PHPDoc to `bool` and then phpstan would not complain. I'm fine with both approaches.

Reply:
If we keep `@return` boolean, PHPStan will not complain in the test.

To be honest, I don't really like that we redefined the result with the PHPDoc - it could screw us over in later on.

> IMO if there's a method that theoretically could be tested, but isn't, then that shouldn't be excluded from code coverage. 

Agreed.

Reply:
> To be honest, I don't really like that we redefined the result with the PHPDoc - it could screw us over in later on.

I think it's OK - TBH I'd probably have `isScenarioMatch() : false` if this was a `final` class (which is fine even if the parent interface is `:bool`. These classes are all defined as being filters that apply to features, so I don't see how we'd ever need them to match a scenario.

Reply:
I went for the approach of keep bool return type (I even removed the docblocks since they seem redundant) and added some tests.